### PR TITLE
Allow Members to Opt-In to Active Status

### PR DESCRIPTION
**Pull request by RamZallan**
Implementation of #141

### CODE REVIEW

File: frontend/javascript/modules/becomeActive.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import FetchUtil from "../utils/fetchUtil";
+
+export default class becomActive {
```
Comment (audiolion) [inline frontend/javascript/modules/becomeActive.js]: missing an `e` in `becomActive`

**Comment by mbillow** [comment]
@audiolion When I was Financial I changed the system so that anyone who wanted to become active could, then a list of active members was sent to Housing to be charged. 

I would read #141 if you haven't already.

File: frontend/javascript/modules/becomeActive.js (hunk 1 above)
Comment (RamZallan) [inline frontend/javascript/modules/becomeActive.js]: Fixed, thanks!

File: frontend/javascript/modules/becomeActive.js [hunk 2]
```diff
@@ -20,12 +20,11 @@ export default class becomActive {
     };
 
     FetchUtil.postWithWarning(this.endpoint, payload, {
-      method: 'POST',
       warningText: "Becoming an active member means that you will be charged" +
                     " dues, which are $80 per semester.",
       successText: "You are now an active member."
     }, () => {
-      document.getElementById('coop-' + this.uid).remove();
+      document.getElementById('becomeActive').remove();
```
Comment (mbillow) [inline frontend/javascript/modules/becomeActive.js]: You are only hiding the button, not the entire div.

File: conditional/templates/dashboard.html [hunk 3]
```diff
@@ -251,15 +251,15 @@ <h3 class="panel-title">Major Projects</h3>
...
+            {% if student and not active %}
+                <div class="alert alert-info"><span class="glyphicon glyphicon-exclamation-sign white" style="padding-right:5px"></span>
+                    Hey there! You're eligible to become an active member, click the button below if you'd like to pay dues and become active!
+                    <br />
+                    <a href="#" id="becomeActive" data-module="becomeActive" data-uid="{{username}}" class="btn btn-default">Become Active</a>
+                </div>
+            {% endif %}
 
-        {% if student and not active %}
-            <div class="alert alert-info"><span class="glyphicon glyphicon-exclamation-sign white" style="padding-right:5px"></span>
```
Comment (mbillow) [inline conditional/templates/dashboard.html]: Not sure an alert is the best container for this, I would go with a panel. 

If you really want to bring attention to it, a colored panel it definitely an option.

```
<div class="panel panel-info">
    <div class="panel-heading">
        <h3 class="panel-title">Panel info</h3>
    </div>
    <div class="panel-body">
        Panel content
    </div>
</div>
```

File: conditional/templates/dashboard.html [hunk 4]
```diff
@@ -251,15 +251,15 @@ <h3 class="panel-title">Major Projects</h3>
             </div>
             {% endif %}
 
-        </div>
+            {% if student and not active %}
+                <div class="alert alert-info"><span class="glyphicon glyphicon-exclamation-sign white" style="padding-right:5px"></span>
+                    Hey there! You're eligible to become an active member, click the button below if you'd like to pay dues and become active!
```
Comment (mbillow) [inline conditional/templates/dashboard.html]: Hey there, you're eligible to become an active member! Click the button below if you'd like to become active and pay dues.

File: conditional/blueprints/member_management.py [hunk 5]
```diff
@@ -603,6 +630,32 @@ def clear_active_members():
...
+    active_list = [["Full Name", "RIT Username", "Amount to Charge"]]
+    for member in ldap_get_active_members():
+        full_name = member.cn
+        rit_username = re.search(".*uid=(\\w*)", member.ritDn).group(1)
+        will_coop = CurrentCoops.query.filter(
+            CurrentCoops.date_created > start_of_year(),
+            CurrentCoops.semester != "Neither",
+            CurrentCoops.uid == member.uid).first()
+        if will_coop:
+            dues = 80
```
Comment (csssuf) [inline conditional/blueprints/member_management.py]: It might be nice to make the price of dues a site setting too, but I wouldn't consider it a blocker.

Comment (RamZallan) [inline conditional/blueprints/member_management.py]: Definitely agree, but should probably be in a separate PR/GH Issue

File: conditional/blueprints/member_management.py [hunk 6]
```diff
@@ -603,6 +630,32 @@ def clear_active_members():
...
+    sio = io.StringIO()
+    csvw = csv.writer(sio)
+
+    active_list = [["Full Name", "RIT Username", "Amount to Charge"]]
+    for member in ldap_get_active_members():
+        full_name = member.cn
+        rit_username = re.search(".*uid=(\\w*)", member.ritDn).group(1)
+        will_coop = CurrentCoops.query.filter(
+            CurrentCoops.date_created > start_of_year(),
+            CurrentCoops.semester != "Neither",
```
Comment (csssuf) [inline conditional/blueprints/member_management.py]: Is this value possible? [models.py](https://github.com/ComputerScienceHouse/conditional/blob/develop/conditional/models/models.py#L196) doesn't seem to indicate it is.

Comment (RamZallan) [inline conditional/blueprints/member_management.py]: I don't think it is, now that you mention it. I was modeling my query after [this one in the /manage route](https://github.com/ComputerScienceHouse/conditional/blob/develop/conditional/blueprints/member_management.py#L72), where they use that. I'll fix this query in a new commit

File: conditional/blueprints/member_management.py (hunk 5 above)
Comment (mbillow) [inline conditional/blueprints/member_management.py]: This is probably 5-10 minutes of work, I don't see why it shouldn't be included. You just have to add an option in the config and then import the `app.config` into this file.

File: conditional/blueprints/member_management.py (hunk 6 above)
Comment (mbillow) [inline conditional/blueprints/member_management.py]: Yah, `Neither` was initially going to be a part of the scheme, but it was removed when we talked about how the system should work modification-wise.

**Comment by mbillow** [comment]
@RamZallan In my initial testing, everything looks to work as intended. Just squash those fix commits (d9fa383 / bd629ae) and we should be good to merge this.
